Description: Fix bytes vs strings in setup.py
Author: Axel Beckert <abe@debian.org>
Forwarded: no

Index: wicd/setup.py
===================================================================
--- wicd.orig/setup.py	2019-09-29 16:21:49.707310027 +0200
+++ wicd/setup.py	2019-09-29 16:22:28.300466696 +0200
@@ -636,7 +636,7 @@
                         print(len(output), returncode)
                         raise ValueError
                     else:
-                        m = re.match(b'(\d+) translated messages(?:, (\d+) fuzzy translation)?(?:, (\d+) untranslated messages)?.', output)
+                        m = re.match('(\d+) translated messages(?:, (\d+) fuzzy translation)?(?:, (\d+) untranslated messages)?.', output)
                         if m:
                             done, fuzzy, missing = m.groups()
                             fuzzy = int(fuzzy) if fuzzy else 0
